include/deal.II/base/parameter_handler.h

```diff
@@ -2230,6 +2258,20 @@ ParameterHandler::add_parameter(const std::string &          entry,
   add_action(entry, action);
 }
 
+/**
+ * Free function that allows to parse input from a specified @param parameter_file
+ * independently of the type of input file (prm, xml, json) being used.
+ * The user has to specify whether parameters in the input file not added to the
+ * parameter handler will be skipped (enables partial parsing), and whether the
+ * code will assert that all parameters of the parameter handler are indeed
+ * found in the input file.
+ */
+void
+parse_input(std::string const parameter_file,
+            ParameterHandler &prm,
+            bool const        skip_undefined               = true,
+            bool const        assert_all_entries_are_found = true);
```

Comment:
Like I said before, let's move this into a separate PR and focus on one aspect at a time. I'm still of the opinion that this should be a member function :-)

(Why? Among other reasons because global functions are hard to find in the documentation. Another reason is that changes in the internal implementation of the `ParameterHandler` class's data structures and public interface can require adjustments in functions; for member functions, these are easy to make, but for global functions they are not.)